lib/fern/api.rb

```diff
@@ -8,14 +8,43 @@ module Api
 
     included do
       class_eval { class_attribute :fern }
-      self.fern = {}
-      def present(obj)
-        render json: obj
+
+      def form
+        if @form.nil?
+          p = params
+          if fern[action_name.to_sym][:form].has_key?(:key) && fern[action_name.to_sym][:form][:key]
+            p = params[fern[action_name.to_sym][:form][:key]]
+          end
+          p = p&.to_hash&.deep_symbolize_keys
+          @form ||= fern[action_name.to_sym][:form][:klass].new(p)
+        end
+        @form
+      end
+
+      def present(object, **opts)
+        render json: render_presenter(object, **opts)
+      end
+
+      def render_presenter(object, **opts)
+        if object.is_a?(Array) || object.is_a?(ActiveRecord::Relation)
```

Comment:
I wonder if there's a better way to do this. Maybe `respond_to?(:map)`? What we really care about here is whether or not `object` quacks like a list.

Reply:
I'm not sure what this is about. It looks to be supporting something like `present multiple_items`, but the only time we really present multiple objects is with a `SearchResultsPresenter`, which actually doesn't trigger this conditional because `present results...` is a single `Hash`, not an Array or Relation.

In the `SearchResults...` case, your suggestion wouldn't work because `Hash` does respond to map (but we don't want to call a new presenter for each item in the Hash).